selectors/selectors.md

```diff
@@ -117,9 +117,14 @@ type ExploreRange struct {
 ## Be careful when using ExploreRecursive with a large maxDepth parameter;
 ## it can easily cause very large traversals (especially if used in combination
 ## with selectors like ExploreAll inside the sequence).
+##
+## If no maxDepth is specified, recursion will continue as long as
+## the selection continues to traverse ExploreRecursiveEdge - be very careful
```

Comment:
In the case that the implementation has its own maxDepth for protection, the semantics wouldn’t be that not setting it traverses indefinitely, it would be that it just fails once it hits a specific depth.

This implies that, if the implementation has its own maxDepth, it should fail when no maxDepth is sent since it will not support an infinite recursion. I don’t think that’s the intended behavior though, I think the intended behavior is probably for recursion to go until it reaches the maxDepth it set as a default.